linera-core/src/client/mod.rs

```diff
@@ -929,16 +929,18 @@ where
         &self,
         committee: &Committee,
         certificate: ValidatedBlockCertificate,
-    ) -> Result<Certificate, ChainClientError> {
+    ) -> Result<ConfirmedBlockCertificate, ChainClientError> {
         let hashed_value =
             HashedCertificateValue::new_confirmed(certificate.executed_block().clone());
         let finalize_action = CommunicateAction::FinalizeBlock {
             certificate: certificate.into(),
             delivery: self.options.cross_chain_message_delivery,
         };
-        let certificate = self
+        let certificate: ConfirmedBlockCertificate = self
             .communicate_chain_action(committee, finalize_action, hashed_value)
-            .await?;
+            .await?
+            .try_into()
+            .expect("Finalized block should be confirmed");
```

Comment:
(I wonder if we'll have to split up `communicate_chain_action` in the end. :thinking: Or make it generic somehow?)

Reply:
There are a couple of API endpoints that either accept an old `Certificate` enum or return one. It'll require either splitting it down into 3 endpoints now or make it generic somehow. I don't have any idea about that yet, I was going through the "easy" parts for now.